src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1499,28 +1500,38 @@ ExecuteModifyTasks(List *taskList, bool expectResults, ParamListInfo paramListIn
 				SetCitusNoticeLevel(INFO);
 			}
 
-			/*
-			 * If caller is interested, store query results the first time
-			 * through. The output of the query's execution on other shards is
-			 * discarded if we run there (because it's a modification query).
-			 */
-			if (placementIndex == 0 && expectResults)
+			PG_TRY();
```

Comment:
`ConsumeQueryResult()` might raise errors with multiple `errcontext()`s. If we wanted to change DefferedErrors to support this, change would be huge, so I am using a `PG_TRY()` block instead.

Reply:
Are there other execution paths we should consider doing this for? e.g. COPY,

Reply:
Added `COPY`, still looking for other execution paths ...